src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -1005,6 +1092,25 @@ public ValueTask SendAsyncForNetworkStream(Socket socket, CancellationToken canc
                     ValueTask.FromException(CreateException(error));
             }
 
+            public ValueTask SendPacketsAsync(Socket socket, CancellationToken cancellationToken)
+            {
+                Debug.Assert(Volatile.Read(ref _continuation) == null, "Expected null continuation to indicate reserved for use");
+
+                if (socket.SendPacketsAsync(this))
```

Comment:
Would be good to add a comment re cancellation here; e.g. "TODO: support cancellation by passing cancellationToken down through SendPacketsAsync etc